src/gridstack.ts

```diff
@@ -1677,17 +1621,27 @@ export class GridStack {
     return this;
   }
 
-  /** @internal call to write position x,y,w,h attributes back to element */
-  protected _writePosAttr(el: HTMLElement, n: GridStackPosition): GridStack {
-    if (n.x !== undefined && n.x !== null) { el.setAttribute('gs-x', String(n.x)); }
-    if (n.y !== undefined && n.y !== null) { el.setAttribute('gs-y', String(n.y)); }
-    n.w > 1 ? el.setAttribute('gs-w', String(n.w)) : el.removeAttribute('gs-w');
-    n.h > 1 ? el.setAttribute('gs-h', String(n.h)) : el.removeAttribute('gs-h');
+  /**
+   * Call to write position x,y,w,h attributes back to element
+   * In addition, updates the inline top/height inline style as well
+   * @internal
+   */
+  protected _writePosAttr(el: HTMLElement, node: GridStackNode): GridStack {
+    if (node.x !== undefined && node.x !== null) { el.setAttribute('gs-x', String(node.x)); }
+    if (node.y !== undefined && node.y !== null) { el.setAttribute('gs-y', String(node.y)); }
+    node.w > 1 ? el.setAttribute('gs-w', String(node.w)) : el.removeAttribute('gs-w');
+    node.h > 1 ? el.setAttribute('gs-h', String(node.h)) : el.removeAttribute('gs-h');
+    // Avoid overwriting the inline style of the draggable element, but update the placeholder
+    if (!node._moving || this._placeholder === el) {
+      // Set inline style, refer CSS variables

```

Comment:
ok, so instead of having global styles at 1617 (written once) you now have 2 inline styles (and 2 more for column if we did the same) PER instance and you think that is actually faster for large number of items ?
I think the opposite.

Also are the var used as is or actual values ? meaning if I change cellHeight so we need to update ALL items styles all over again ? just making sure it write el.style.top = calc(1 * var(--gs-cell-height))

Also technically we don't need gs-y | gs-h anymore since they don't refer to styles now so we could remove.
and you need to remove style when h=1 like I did for gs-h

Reply:
The formula is the same, don't need to update all nodes when the cell-height changes. The CSS vars will be recalculated by the browser when the var changes.
Using inline style (even if this is a formula like that) is O(N) where N = number of widgets in the gridstack. Instead, using row/column rules is O(R+C) where R and C are the number of rows and columns, regardless the nodes actually on screen.
Using a finer resolution in the grid causes more rules to be put in CSS, even if only one node is on screen.

Given the intrinsic performance used by displaying nodes, having a O(N) rule in this case is more linear to the screen DOM complexity.

> Also technically we don't need gs-y | gs-h anymore since they don't refer to styles

I'm not sure if such attributes are used by the library or by other applications to do some logic, so probably it is better to leave them to avoid regressions.

Reply:
actually O(N) vs O(R!) - really 2xN vs 2xR! but yeah as I've seen people use finner resolutions, even down to pixels so this then makes a lot more sense.